backend/src/ml_space_lambda/group/lambda_functions.py

```diff
@@ -121,11 +125,31 @@ def group_datasets(event, context):
 
 @api_wrapper
 def add_users(event, context):
+    env_variables = get_environment_variables()
     group_name = event["pathParameters"]["groupName"]
     request = json.loads(event["body"])
     usernames = request["usernames"]
-    for username in usernames:
-        _add_group_user(group_name, username, [Permission.COLLABORATOR])
+    if group_dao.get(group_name):
+        project_groups = project_group_dao.get_projects_for_group(group_name)
+        for username in usernames:
+            if user_dao.get(username):
+                group_user_dao.create(
+                    GroupUserModel(
+                        group_name=group_name,
+                        username=username,
+                        permissions=[Permission.COLLABORATOR],
+                    )
+                )
+
+                if env_variables[EnvVariable.MANAGE_IAM_ROLES]:
+                    for project_group in project_groups:
+                        iam_role_arn = iam_manager.get_iam_role_arn(project_group.project, username)
+                        if iam_role_arn is None:
+                            iam_role_arn = iam_manager.add_iam_role(project_group.project, username)
+
+                iam_manager.update_user_policy(username)
+            else:
+                raise Exception("Username specified is not associated with an active user.")
```

Comment:
does raising an exception here halt execution for the other users? 

This might be fine if we want this operation to be atomic, but it could also be annoying if the user just wants to add valid users and be alerted that at least one user failed to be added. I think we should at least raise the exception with the offending user's username so the error displayed in MLS lets the admin know which user is causing problems.

Reply:
You're right it would. I just folded back in [this code](https://github.com/awslabs/mlspace/pull/246/files/55a3316b3d5d6fa3e4149c5a242ed99221433afe#diff-cc7e87a7a0f5f44978d1b8428b06133c092432c8cd6b435298518025bc2e2042L46-L60) so we had this same behavior before but it isn't a great customer experience. I think the correct course of action is to validate all the users exist first and only then do the updates. This way we're at least trying to avoid putting things into a partial state and we can inform the requester what users are problematic.